yt_dlp/extractor/xumo.py

```diff
@@ -0,0 +1,155 @@
+                            'ext': determine_ext(sub_url),
+                            'url': sub_url
+                        }]
+                    }
+                    subtitles = self._merge_subtitles(subtitles, subs)
+
+        if not formats and has_drm:
+            self.report_drm(video_id)
+        self._sort_formats(formats)
+
+        return formats, subtitles
+
+    def _real_extract(self, url):
+
+        mobj = self._match_valid_url(url).groupdict()
+        media_id = mobj['id']
+        media_type = mobj['mediaType']
+
+        webpage = self._download_webpage(url, media_id)
+
+        page_data = self._search_json('<script id="__NEXT_DATA__" type="application/json">', webpage, media_id, media_id)
+        page_props = traverse_obj(page_data, ('props', 'pageProps'))
+        entity = traverse_obj(page_props, ('page', 'entity'))
+        thumbnail = entity.get('image') + '/1024x576' + entity.get('imageExt')
+
+        title = entity.get('title')
+
+        if media_type == 'free-movies':
+            # movie => return video info
+            formats, subtitles = self._get_video_info(media_id, entity)
+            return {
+                'id': media_id,
+                'title': title,
+                'description': entity.get('description'),
+                'release_year': entity.get('year'),
+                'duration': entity.get('duration'),
+                'thumbnail': thumbnail,
+                'formats': formats,
+                'subtitles': subtitles,
+            }
+        elif media_type == 'tv-shows':
+            # series => return set of video info for each episode
+            videos = []
+            for season in entity['seasons'] or []:
+                season_number = traverse_obj(season, ('season', 'number'))
+                season_name = traverse_obj(season, ('category', 'name'))
+                for episode in season['cards'] or []:
+                    episode_id = episode['id']
+                    episode_thumb = episode.get('image') + '/800x450' + episode.get('imageExt')
+                    video_json = self._download_json(self._INFO_URL + episode_id + '.json', episode_id, query=self._INFO_QUERY_PARAMS)
+                    formats, subtitles = self._get_video_info(episode_id, video_json)
+                    videos.append({
+                        'id': episode_id,
+                        'title': episode.get('title'),
+                        'description': episode.get('description'),
+                        'duration': episode.get('duration'),
+                        'thumbnail': episode_thumb,
+                        'season': season_name,
+                        'episode': episode.get('title'),
+                        'season_number': int(season_number),
+                        'episode_number': int(episode.get('episode')),
+                        'formats': formats,
+                        'subtitles': subtitles,
+                        'series': title
+                    })
```

Comment:
Why is single episodes not supported?

Reply:
When you navigate to a series from the TV shows tab, selecting an episode doesn't give you an episode specific URL (stays on the series URL). I'm not sure if there's a straightforward way for non-developers to reliably get a URL for a specific episode.

Reply:
Even so, support it. And return appropriate url_result for playlists. This method creates problems with some options like --flat-playlist

Reply:
Support for individual episodes has now been added.